usaspending_api/awards/migrations/0076_auto_20200916_0312.py

```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.2.16 on 2020-09-16 03:12
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('awards', '0075_financialaccountsbyawards_distinct_award_key'),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name='financialaccountsbyawards',
+            index=models.Index(condition=models.Q(disaster_emergency_fund__in=['L', 'M', 'N', 'O', 'P']), fields=['submission', 'distinct_award_key', 'transaction_obligated_amount', 'gross_outlay_amount_by_award_cpe'], name='faba_subid_awardkey_sums_idx'),
```

Comment:
One other thing that came to my attention when looking at QAT is that I think the counts of awards for CFDA uses a different query, that may not benefit as much from the `distinct_award_key` or this index. Because it first tries to narrow down to "All Assistance Awards". I think it does that by checking FAIN or URI is not null. @tony-sappe can you clarify how that query is run and how it's alike or different from the one for "All Awards" in its use of award identifiers?

If that's the case, and it is isolating on "NULLness" of FAIN/URI, should we add those to this index? Or add another partial index similar to this that has a partial clause about whether those are NULL?

Reply:
@kwhickey the query to count assistance awards check if `piid` is null or not-null

Reply:
Thanks. I had it backwards. 
@tony-sappe is it using `distinct_award_key`, like?
```sql
SELECT ..., count(distinct distinct_award_key)
WHERE faba.piid IS NULL
GROUP BY faba.distinct_award_key
```
I'm wondering if that query will or will not use this index. And if that query could be much more improved if a partial index was added with condition of `piid IS NULL and disaster_emergency_fund_code IN ('L','M','N','O','P')`

Reply:
I don't see how that query using `piid` can use the new index for an **index-only scan**. It could still use the index depending on what the planner determines to be the best path.